### DISCUSSION TITLE
Make base URL configurable

### DISCUSSION THREAD

**Discussion by buhrmi** [inertiajs/inertia#934] [Ideas]
Currently, Inertia always uses `window.location` as the base when constructing URLs ([see here](https://github.com/inertiajs/inertia/blob/master/packages/inertia/src/url.ts#L6)). I would like to make this configurable so that I can use Inertia in the context of a mobile app.

The app is packaged with Capacitor and delivered through the app store. When the app launches, I make a mock Inertia request to my server to get the initial page object (instead of taking it from the `data-page` attribute of the `#app` element). Now I would like to tell Inertia to make requests to my server instead of whatever `window.location` is.

**Comment by buhrmi** [comment]
Now I'm thinking that adding a configuration option like ` createInertiaApp({endpoint: 'https://someurl', ...}) ` would do the trick.

**Comment by ajnsn** [comment]
The second parameter of the [URL() constructor](https://developer.mozilla.org/en-US/docs/Web/API/URL/URL) only applies when the `url` is a relative url.

You can use absolute URLs to bypass your issue? 
Demo:

```js
function hrefToUrl(href) {
  return new URL(href.toString(), 'https://inertiajs.com')
}

hrefToUrl('routing'); // https://inertiajs.com/routing

hrefToUrl('https://github.com/inertiajs/inertia'); // https://github.com/inertiajs/inertia
```

**Reply by buhrmi** [reply]
that's a good idea, didn't think of that. yeah, that should work.
